test/functional/mempool_unbroadcast.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test that the mempool ensures transaction delivery by periodically sending
+to peers until a GETDATA is received."""
+
+from io import BytesIO
+import time
+
+from test_framework.blocktools import create_block, create_coinbase
+from test_framework.messages import ToHex, CTransaction
+from test_framework.mininode import P2PTxInvStore
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+        assert_equal,
+        disconnect_nodes,
+        connect_nodes,
+        create_confirmed_utxos,
+        wait_until,
+        hex_str_to_bytes
+)
+
+class MempoolUnbroadcastTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        self.test_broadcast()
+        self.test_txn_removal()
+
+    def test_broadcast(self):
+        self.log.info("Test that mempool reattempts delivery of locally submitted transaction")
+        node = self.nodes[0]
+
+        min_relay_fee = node.getnetworkinfo()["relayfee"]
+        utxos = create_confirmed_utxos(min_relay_fee, node, 10)
+
+        disconnect_nodes(node, 1)
+
+        self.log.info("Generate transactions that only node 0 knows about")
+        # generate a wallet txn
+        addr = node.getnewaddress()
+        wallet_tx_hsh = node.sendtoaddress(addr, 0.0001)
+
+        # generate a txn using sendrawtransaction
+        us0 = utxos.pop()
+        inputs = [{ "txid" : us0["txid"], "vout" : us0["vout"]}]
+        outputs = {addr: 0.0001}
+        tx = node.createrawtransaction(inputs, outputs)
+        node.settxfee(min_relay_fee)
+        txF = node.fundrawtransaction(tx)
+        txFS = node.signrawtransactionwithwallet(txF['hex'])
+        rpc_tx_hsh = node.sendrawtransaction(txFS['hex'])  # txhsh in hex
+
+        # check that second node doesn't have these two txns
+        mempool = self.nodes[1].getrawmempool()
+        assert(rpc_tx_hsh not in mempool)
+        assert(wallet_tx_hsh not in mempool)
```

Comment:
In commit f40f6875f5570cb25882d8a04a6cbeb61ee7366d:

I think assert is a statement, not a function, so should be without `()`
```suggestion
        assert wallet_tx_hsh not in mempool
```

Reply:
@amitiuttarwar Same goes for this one: `black` on the entire file to get proper formatting for free :)

Reply:
fixed